contracts/contracts/coordination/SubscriptionManager.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "./ManagedAllowList.sol";
+
+contract SubscriptionManager is ManagedAllowList {
+    struct Subscription {
+        address subscriber;
+        uint256 amount;
+        uint256 period; // Represents the end timestamp of the subscription
+    }
+
+    mapping(bytes32 => Subscription) public subscriptions;
+
+    event SubscriptionCreated(
+        uint32 indexed ritualId,
+        address indexed subscriber,
+        uint256 amount,
+        uint256 period
+    );
+    event SubscriptionTerminated(uint32 indexed ritualId, address indexed subscriber);
+    event SubscriptionRenewed(
+        uint32 indexed ritualId,
+        address indexed subscriber,
+        uint256 amount,
+        uint256 period
+    );
+
+    function createSubscription(
+        uint32 ritualId,
+        address _subscriber,
+        uint256 _amount,
+        uint256 _period
+    ) external onlyCohortAuthority(ritualId) {
+        bytes32 key = lookupKey(ritualId, _subscriber);
+        require(
+            subscriptions[key].subscriber == address(0),
+            "Subscriber already has an active subscription"
+        );
+
+        Subscription memory newSubscription = Subscription({
+            subscriber: _subscriber,
+            amount: _amount,
+            period: _period
+        });
```

Comment:
that is a bit ineffective, creating memory variable and then copy it to storage. Would be better just fill up storage variable like in renew method: 
```
Subscription storage newSubscription = subscriptions[key]; 
newSubscription.subscriber = _subscriber;
...
```